### PR TITLE
[#1341] Hash sort order in commits panel

### LINKED ISSUE #1341: Hashing of settings in commits panel
> Hashing of filters (and any other setting in the page) could be useful as we want to be able to share a report exactly as seen by the viewer by simply sending the current URL to someone else. But we should do it in a way that can be applied to all pages rather than just the commits page.

Originally posted by @damithc in #1239 
It would be good if we can hash the settings of commits panel for the above mentioned reasons.
Potential settings that can be hashed includes sort by attribute, sorting order, hiding of commit messages.

### PR DESCRIPTION
**Pull request by Boon-Jun**
```
Currently, commits panel already supports the hashing of several attributes
(e.g fileTypes, Author, Repo etc.), but not the sorting order of commits.

To allow the sharing of sort settings via the URL, and also for completeness, 
let's hash the sorting order of commits in the commits-panel.
```

Resolves #1341

### CODE REVIEW

**Comment by Boon-Jun** [comment]
Hi all, I'll have to wrap up this PR before a module deadline this Sunday. Any comments on are greatly appreciated!

File: frontend/cypress/tests/zoomView/zoomView_reload.js [hunk 1]
```diff
@@ -9,10 +9,10 @@ describe('reload page', () => {
         .click();
 
     cy.get('#tab-zoom > .sorting > .sort-by > select:visible')
-        .select('Time');
+        .select('LoC');
```
Comment (fzdy1914) [inline frontend/cypress/tests/zoomView/zoomView_reload.js:12]: Why the default controls will be affected here? I didn't see the related code change?

**Comment by 0blivious** [comment]
Restoring not working correctly
Try open zoom panel -> select sort by LOC + Ascending -> refresh the page

File: frontend/cypress/tests/zoomView/zoomView_reload.js (hunk 1 above)
Comment (Boon-Jun) [reply frontend/cypress/tests/zoomView/zoomView_reload.js:12]: The default controls are not affected here. But the purpose of selecting LoC is to ensure that it remains selected upon reload.

File: frontend/src/static/js/v_zoom.js [hunk 2]
```diff
@@ -145,6 +153,22 @@ window.vZoom = {
...
+    },
+
+    retrieveSortHash() {
+      window.decodeHash();
+      const hash = window.hashParams;
+      if (hash.zCST) {
+        this.commitsSortType = hash.zCST;
+      }
+      if (hash.zRSC) {
+        this.toReverseSortedCommits = (hash.zRSC === 'true');
```
Comment (Tejas2805) [inline frontend/src/static/js/v_zoom.js:168]: Why do we need the `hash.zRSC === 'true'` ?

Comment (Boon-Jun) [reply frontend/src/static/js/v_zoom.js:168]: hashes are stored as a string, I did this to convert the string into a boolean

**Review by 0blivious — APPROVED** [review]

**Review by jamessspanggg — APPROVED** [review]

File: frontend/cypress/tests/zoomView/zoomView_reload.js (hunk 1 above)
Comment (fzdy1914) [reply frontend/cypress/tests/zoomView/zoomView_reload.js:12]: OK, understood

**Pull request merged**